tests/test_preconditions.py

```diff
@@ -20,35 +21,58 @@ def start_gprofiler(
     privileged: bool = True,
     user: int = 0,
     pid_mode: Optional[str] = "host",
+    extra_profiler_args: Optional[List[str]] = None,
 ) -> Container:
+    extra_profiler_args = extra_profiler_args or []
     return start_gprofiler_in_container_for_one_session(
         docker_client,
         gprofiler_docker_image,
         Path("/tmp"),
         Path("/tmp/collapsed"),
         [],
-        ["-d", "1"],
+        ["-d", "1"] + extra_profiler_args,
         privileged=privileged,
         user=user,
         pid_mode=pid_mode,
     )
 
 
+@contextmanager
+def run_gprofiler_container(
```

Comment:
This feels like `run_gprofiler_in_container_for_one_session` without the waiting part, no? Do you want to try merging them?

Reply:
No reason, only needed here, all the other tests are waiting for the container to exit, this is only needed when running in continuously

Reply:
Ok